templates/CRM/Event/Form/Registration/DisplayProfile.tpl

```diff
@@ -12,30 +12,59 @@
         <div class="crm-group participant_info-group">
       <div class="header-dark">{if $addParticipantProfile}{ts}Participant 1{/ts}{else}{ts}Participant Information{/ts}{/if}</div>
             {if $primaryParticipantProfile.CustomPre}
-               <fieldset class="label-left no-border"><div class="bold crm-profile-view-title">{$primaryParticipantProfile.CustomPreGroupTitle}</div>
+              {assign var="hasValue" value=false}
+              {foreach from=$primaryParticipantProfile.CustomPre item=value key=field}
+                {if $value !== ''}
+                  {assign var="hasValue" value=true}
+                  {break} <!-- exit if a value is found -->
```

Comment:
`{break}` crashes in smarty 2. I think we're still officially still supporting people who haven't switched yet. Performance-wise given the context I think it would be fine to just leave out the `{break}`.